server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -215,8 +215,17 @@ public ClusterState execute(ClusterState currentState) {
                         final RepositoryMetadata updatedMetadata;
                         if (canUpdateInPlace(newRepositoryMetadata, existing)) {
                             if (repositoryMetadata.settings().equals(newRepositoryMetadata.settings())) {
-                                // Previous version is the same as this one no update is needed.
-                                return currentState;
+                                if (repositoryMetadata.generation() == RepositoryData.CORRUPTED_REPO_GEN) {
```

Comment:
I think it would be nice to have a log message indicating that we are recreating a repository marked as corrupted (the message should contain the name and the known UUID of the repository)